### ISSUE TITLE
Upload answers to bad input using a `redirect(400)`

### ISSUE THREAD

**Issue by felix91gr** [azlux/botamusique#146]
**Describe the bug**

The bug happens here:

https://github.com/azlux/botamusique/blob/master/interface.py#L569-L619

Every time that the program uses `redirect` like this:
* `redirect("./", code=400)`
* `redirect("./", code=500)`

It does not work 100% as intended.

The problem is two-fold:
1. It doesn't actually redirect! After some testing, I've come to the conclusion that this is because we're using `4xx` and `5xx` codes with the `redirect` function, which only is documented to support `301, 302, 303, 305, 307` and `308` https://werkzeug.palletsprojects.com/en/1.0.x/utils/. I tested the `400` calls with a `304` code and they don't redirect either. However, `303` works flawlessly. This seems to be an intended behaviour of `werkzeug`, and that makes a lot of sense, because...
2. Redirect responses are only `3xx`. However we're trying to redirect with a `400` and `500` response. We should maybe be using a different HTTP response other than `redirect`, because what we're trying to convey are errors.

The following responses I would keep as they are, but would change from `redirect` to something else (I think `werkzeug` has specific functions for them): 
* https://github.com/azlux/botamusique/blob/master/interface.py#L575 responds to `files` not existing. This should be `400` I think, since the request is clearly malformed.
* https://github.com/azlux/botamusique/blob/master/interface.py#L604 should still be `500`, because it is trying to handle the scenario in which the server simply failed to create the file for one reason or another.

The following responses I would change from `400` and would also not use with `redirect`:
* https://github.com/azlux/botamusique/blob/master/interface.py#L581 responds to the `filename` being empty. This should be a `422` I think, because the request is well-formed but its contents are semantically incorrect.
* https://github.com/azlux/botamusique/blob/master/interface.py#L587 responds to the `directory` being something above the root folder. This should also be a `422` if the prior one is `422`, since this request is also well-formed but the contents are semantically incorrect.
* https://github.com/azlux/botamusique/blob/master/interface.py#L598 is also a `422` candidate

**Affected version**

Tag `6.3`

**To Reproduce**

There are many HTTP responses in this bug report. However, the following one is very simple:

1. Open the bot's page
2. Click on "upload" without selecting a file
3. Response will be `400`
4. The page wants to redirect, but can't, because `werkzeug` will not redirect under code `400`

**Expected behavior**

The page says it wants to redirect. At first glance, I thought it should've. But now that I have taken a closer look at it, I think it's mostly okay - it should just not try to redirect.

**Desktop (please complete the following information):**
 - OS: Windows and Ubuntu
 - Browser: Firefox and Chrome
 - Version: Latest (Ubuntu is LTS)

**Comment by TerryGeng** [comment]
Thanks for your advice! I have to admit, the HTTP error code we are using here is a bit the result of my whimsical choice and need to be reviewed at some point. I will start a PR about this issue, and you are welcomed to review it :3

**Comment by TerryGeng** [comment]
I think I can close this issue now. Do you agree? @felix91gr

**Comment by felix91gr** [comment]
Ohh! I just saw that commit. That looks awesome.

During the weekend we had an issue with the internet of the place where our server runs, so our Mumble is currently offline. But what I see in the code looks excellent :3 specially the handling of the HTTP requests. I think that's 10/10, it aborts on 4xx and all :heart: 

Since the original problem that gave the title to this issue has been solved, I think we can close it.
If the upload has some other problem now, it will definitely not be a `redirect(400)`. And if we see such a problem, we'll open another issue to address that ^^.

Thank you @TerryGeng :rainbow:
